packages/core/database/lib/entity-manager/regular-relations.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+const {
+  isBidirectional,
+  isOneToAny,
+  isManyToAny,
+  isAnyToOne,
+  isAnyToMany,
+} = require('../metadata/relations');
+const { createQueryBuilder } = require('../query');
+
+const getSelect = (joinTable, attribute) => {
+  const { joinColumn, orderColumnName, inverseJoinColumn, inverseOrderColumnName } = joinTable;
+  const select = [joinColumn.name, inverseJoinColumn.name];
+  if (isAnyToMany(attribute)) {
+    select.push(orderColumnName);
+  }
+  if (isBidirectional(attribute) && isManyToAny(attribute)) {
+    select.push(inverseOrderColumnName);
+  }
+  return select;
+};
+
+const deletePreviousOneToAnyRelations = async ({ id, attribute, joinTable, relIdsToadd, db }) => {
+  const { joinColumn, inverseJoinColumn, orderColumnName } = joinTable;
+  const select = getSelect(joinTable, attribute);
+
+  // need to delete the previous relations for oneToAny relations
+  if (isBidirectional(attribute) && isOneToAny(attribute)) {
+    // update orders for previous oneToAny relations that will be deleted if it has order (oneToMany)
+    if (isAnyToMany(attribute)) {
+      const currentRelsToDelete = await createQueryBuilder(joinTable.name, db)
+        .select(select)
+        .where({
+          [inverseJoinColumn.name]: relIdsToadd,
+          [joinColumn.name]: { $ne: id },
+        })
+        .where(joinTable.on || {})
+        .execute();
+
+      currentRelsToDelete.sort((a, b) => b[orderColumnName] - a[orderColumnName]);
```

Comment:
again, same question as before, why not ordering it in the query above?

Reply:
good idea 👌